oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/introspection/OAuth2TokenIntrospectionClient.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.introspection;
+
+import java.util.Map;
+
+/**
+ * Implementations of this interface are responsible for &quot;introspecting&quot;
+ * the contents of an OAuth 2.0 Token.
+ *
+ * A typical implementation would contact an
+ * <a href="https://tools.ietf.org/html/rfc7662" target="_blank">OAuth 2.0 Introspection Endpoint</a>
+ * hosted on an OAuth 2.0 Authorization Server.
+ *
+ * @author Josh Cummings
+ * @since 5.2
+ */
+public interface OAuth2TokenIntrospectionClient {
+
+	/**
+	 * Introspect the given token's attributes
```

Comment:
I like the class level Javadoc, but just looking at this method the description is a bit abstract to me. Perhaps you can provide more details or a link to the class level Javadoc? 

Whatever you do here, should also be done on the reactive bits.

Reply:
@rwinch I've updated the method-level Javadoc - https://github.com/spring-projects/spring-security/pull/6633/commits/dd2f331f5d435b5bbc51b2c8beb6dac2583f387b

Let me know if that's more in line with what you were thinking.